adapter-static `fallback` page not generated when `prerender.enabled` is `false`

**Describe the bug**
Using the adapter static in "SPA" configuration doesn't output any html entry point even though the build completes without errors. All other bundled assets are created as far as i can tell. `/build` contains

```
/_app
  |--/assets
  |--/chunks
  |--/pages
  |--...misc.js
_redirects
```

**Logs**
Build completes successfully with logs ending in 
```
vite v2.3.3 building SSR bundle for production...
✓ 38 modules transformed.
.svelte-kit/output/server/app.js   396.41kb

Run npm run preview to preview your production build locally.

> Using @sveltejs/adapter-static
  ✔ done
✨  Done in 3.72s.
```

**To Reproduce**
svelte.config.js

```js
import preprocess from 'svelte-preprocess';
import adapter from '@sveltejs/adapter-static';

/** @type {import('@sveltejs/kit').Config} */
const config = {
  // Consult https://github.com/sveltejs/svelte-preprocess
  // for more information about preprocessors
  preprocess: preprocess(),
  kit: {
    // hydrate the <div id="svelte"> element in src/app.html
    target: '#svelte',
    adapter: adapter({
      fallback: "index.html"
    }),
    prerender: {
      enabled: false
    },
    ssr: false,
    vite: {
      ssr: {
        noExternal: ["chart.js"]
      },
    }
  }
};

export default config;
```


To help us help you, if you've found a bug please consider the following:

- If possible, we recommend creating a small repo that illustrates the problem.
- Reproductions should be small, self-contained, correct examples – http://sscce.org.

Occasionally, this won't be possible, and that's fine – we still appreciate you raising the issue. But please understand that Svelte is run by unpaid volunteers in their free time, and issues that follow these instructions will get fixed faster.

**Expected behavior**
Entrypoint index html should be generated in the root of the build dir

**Stacktraces**
N/A

**Information about your SvelteKit Installation:**

<details>
  <summary>Diagnostics</summary>

  System:
    OS: macOS 11.1
    CPU: (8) arm64 Apple M1
    Memory: 520.03 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.1.0 - ~/.nvm/versions/node/v16.1.0/bin/node
    Yarn: 1.22.10 - /opt/homebrew/bin/yarn
    npm: 7.11.2 - ~/.nvm/versions/node/v16.1.0/bin/npm
  Browsers:
    Brave Browser: 91.1.25.68
    Firefox: 88.0.1
    Safari: 14.0.2
  npmPackages:
    @sveltejs/kit: ^1.0.0-next.110 => 1.0.0-next.110
    svelte: ^3.34.0 => 3.38.2


- Adapter-Static
</details>

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of SvelteKit entirely?

This is medium high severity, we are unable to deploy our svelte kit SPA, although the dev server works fine

I'm having the same problem as @NickClark. If I turn on prerender a lot of import bugs as he points out:

```> Cannot find module '/home/{user}/www/uitt-svelte-kit/node_modules/highcharts/highcharts-more' imported from /home/{user}/www/uitt-svelte-kit/.svelte-kit/output/server/app.js
Did you mean to import highcharts/highcharts-more.js?```

Same happens with `sveltestrap` that imports `import { createPopper } from '@popperjs/core/dist/esm/popper`, without `.js`.

On the other hand, if I turn off prerendering, then the fallback page is not generated.

--- a/.changeset/grumpy-onions-applaud.md
+++ b/.changeset/grumpy-onions-applaud.md
@@ -1,0 +1,5 @@
+---
+"@sveltejs/kit": patch
+---
+
+[fix] fallback should still be generated when prerender is disabled

--- a/packages/kit/src/core/adapt/prerender.js
+++ b/packages/kit/src/core/adapt/prerender.js
@@ -225,7 +225,7 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 			});
 
 			if (is_html && config.kit.prerender.crawl) {
-				const cleaned = clean_html(/** @type {string} */(rendered.body));
+				const cleaned = clean_html(/** @type {string} */ (rendered.body));
 
 				let match;
 				const pattern = /<(a|img|link|source)\s+([\s\S]+?)>/gm;

--- a/packages/kit/src/core/adapt/prerender.js
+++ b/packages/kit/src/core/adapt/prerender.js
@@ -81,7 +81,8 @@ function chooseErrorHandler(log, onError) {
 const OK = 2;
 const REDIRECT = 3;
 
-/** @param {{
+/**
+ * @param {{
  *   cwd: string;
  *   out: string;
  *   log: Logger;
@@ -89,8 +90,13 @@ const REDIRECT = 3;
  *   build_data: import('types/internal').BuildData;
  *   fallback?: string;
  *   all: boolean; // disregard `export const prerender = true`
- * }} opts */
+ * }} opts
+ */
 export async function prerender({ cwd, out, log, config, build_data, fallback, all }) {
+	if (!config.kit.prerender.enabled && !fallback) {
+		return;
+	}
+
 	__fetch_polyfill();
 
 	const dir = resolve_path(cwd, `${SVELTE_KIT}/output`);
@@ -219,7 +225,7 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 			});
 
 			if (is_html && config.kit.prerender.crawl) {
-				const cleaned = clean_html(/** @type {string} */ (rendered.body));
+				const cleaned = clean_html(/** @type {string} */(rendered.body));
 
 				let match;
 				const pattern = /<(a|img|link|source)\s+([\s\S]+?)>/gm;
@@ -264,6 +270,18 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 		}
 	}
 
+	if (config.kit.prerender.enabled) {
+		for (const entry of config.kit.prerender.pages) {
+			if (entry === '*') {
+				for (const entry of build_data.entries) {
+					await visit(entry, null);
+				}
+			} else {
+				await visit(entry, null);
+			}
+		}
+	}
+
 	if (fallback) {
 		const rendered = await app.render(
 			{
@@ -285,15 +303,5 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 		const file = join(out, fallback);
 		mkdirp(dirname(file));
 		writeFileSync(file, rendered.body || '');
-	} else {
-		for (const entry of config.kit.prerender.pages) {
-			if (entry === '*') {
-				for (const entry of build_data.entries) {
-					await visit(entry, null);
-				}
-			} else {
-				await visit(entry, null);
-			}
-		}
 	}
 }

--- a/packages/kit/src/core/adapt/prerender.js
+++ b/packages/kit/src/core/adapt/prerender.js
@@ -264,16 +264,6 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 		}
 	}
 
-	for (const entry of config.kit.prerender.pages) {
-		if (entry === '*') {
-			for (const entry of build_data.entries) {
-				await visit(entry, null);
-			}
-		} else {
-			await visit(entry, null);
-		}
-	}
-
 	if (fallback) {
 		const rendered = await app.render(
 			{
@@ -295,5 +285,15 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 		const file = join(out, fallback);
 		mkdirp(dirname(file));
 		writeFileSync(file, rendered.body || '');
+	} else {
+		for (const entry of config.kit.prerender.pages) {
+			if (entry === '*') {
+				for (const entry of build_data.entries) {
+					await visit(entry, null);
+				}
+			} else {
+				await visit(entry, null);
+			}
+		}
 	}
 }

--- a/packages/kit/src/core/adapt/utils.js
+++ b/packages/kit/src/core/adapt/utils.js
@@ -31,17 +31,15 @@ export function get_utils({ cwd, config, build_data, log }) {
 		},
 
 		async prerender({ all = false, dest, fallback }) {
-			if (config.kit.prerender.enabled || fallback) {
-				await prerender({
-					out: dest,
-					all,
-					cwd,
-					config,
-					build_data,
-					fallback,
-					log
-				});
-			}
+			await prerender({
+				out: dest,
+				all,
+				cwd,
+				config,
+				build_data,
+				fallback,
+				log
+			});
 		}
 	};
 }

--- a/packages/kit/src/core/adapt/utils.js
+++ b/packages/kit/src/core/adapt/utils.js
@@ -31,7 +31,7 @@ export function get_utils({ cwd, config, build_data, log }) {
 		},
 
 		async prerender({ all = false, dest, fallback }) {
-			if (config.kit.prerender.enabled) {
+			if (config.kit.prerender.enabled || fallback) {
 				await prerender({
 					out: dest,
 					all,

